test/ZNSRootRegistrar.test.ts

```diff
@@ -873,6 +844,55 @@ describe("ZNSRootRegistrar", () => {
       const events = await zns.meowToken.queryFilter(transferEventFilter);
       expect(events.length).to.eq(0);
     });
+
+    it("Sets the payment config when provided with the domain registration", async () => {
+      const tokenURI = "https://example.com/817c64af";
+      const distrConfig : IDistributionConfig = {
+        pricerContract: await zns.curvePricer.getAddress(),
+        paymentType: PaymentType.STAKE,
+        accessType: AccessType.OPEN,
+      };
+
+      await defaultRootRegistration({
+        user,
+        zns,
+        domainName: defaultDomain,
+        tokenURI,
+        distrConfig,
+        paymentConfig: {
+          token: await zns.meowToken.getAddress(),
+          beneficiary: user.address,
+        },
+      });
+
+      const domainHash = hashDomainLabel(defaultDomain);
+      const config = await zns.treasury.paymentConfigs(domainHash);
+      expect(config.token).to.eq(await zns.meowToken.getAddress());
+      expect(config.beneficiary).to.eq(user.address);

```

Comment:
Not sure where I would group this type of test when we rafactor, because we're confirming behaviour on contract that isn't this test file's contract, but is related to it. Probably flow, maybe integration?